python-package/lightgbm/basic.py

```diff
@@ -364,7 +364,7 @@ def _is_2d_collection(data: Any) -> bool:
     )
 
 
-def _is_pyarrow_array(data: Any) -> bool:
+def _is_pyarrow_array(data: Any) -> "TypeGuard[Union[pa_Array, pa_ChunkedArray]]":
```

Comment:
I didn't know about `TypeGuard`, cool stuff!

Reply:
🙌🏻 yeah it's great! I'd never seen it before doing #6106.